src/containerapp/azext_containerapp/containerapp_resiliency_decorator.py

```diff
@@ -0,0 +1,476 @@
+        self.validate_positive_argument("timeout_response_in_seconds")
+        self.validate_positive_argument("timeout_connection_in_seconds")
+        self.validate_positive_argument("tcp_retry_max_connect_attempts")
+        self.validate_positive_argument("http_retry_max")
+        self.validate_positive_argument("http_retry_delay_in_milliseconds")
+        self.validate_positive_argument("http_retry_interval_in_milliseconds")
+
+    def set_up_containerapp_resiliency_yaml(self, file_name):
+        containerapp_def = ContainerAppsResiliencyModel
+        if self.get_argument_tcp_retry_max_connect_attempts() or self.get_argument_circuit_breaker_consecutive_errors()\
+                or self.get_argument_circuit_breaker_interval() or self.get_argument_circuit_breaker_max_ejection() or \
+                self.get_argument_tcp_connection_pool_max_connections() or \
+                self.get_argument_http_connection_pool_http1_max_pending_req() \
+                or self.get_argument_http_connection_pool_http2_max_req() or self.get_argument_timeout_response_in_seconds() \
+                or self.get_argument_timeout_connection_in_seconds() or self.get_argument_http_retry_max() \
+                or self.get_argument_http_retry_delay_in_milliseconds() or self.get_argument_http_retry_interval_in_milliseconds() \
+                or self.get_argument_http_retry_status_codes() or self.get_argument_http_retry_errors():
+            not self.get_argument_disable_warnings() and logger.warning(
+                'Additional flags were passed along with --yaml. These flags will be ignored, and the configuration '
+                'defined in the yaml will be used instead')
+
+        yaml_containerapps_resiliency = load_yaml_file(file_name)
+        if type(yaml_containerapps_resiliency) != dict:  # pylint: disable=unidiomatic-typecheck
+            raise ValidationError('Invalid YAML provided. Please supply a valid YAML spec.')
+
+        if yaml_containerapps_resiliency.get('type') and yaml_containerapps_resiliency.get('type').lower() != "microsoft.app/containerapps/resiliencypolicies":
+            raise ValidationError('Containerapp resiliency type must be \"Microsoft.App/containerApps/resiliencyPolicies\"')
+
+        if yaml_containerapps_resiliency.get('name') and yaml_containerapps_resiliency.get('name').lower() != self.get_argument_name().lower():
+            logger.warning(
+                'The app name provided in the --yaml file "{}" does not match the one provided in the --name flag "{}". The one provided in the --yaml file will be used.'.format(
+                    yaml_containerapps_resiliency.get('name'), self.get_argument_name()))
+            self.set_argument_name(yaml_containerapps_resiliency.get('name'))
+
+        if yaml_containerapps_resiliency.get('containerAppName') and yaml_containerapps_resiliency.get('containerAppName').lower() != self.get_argument_container_app_name().lower():
```

Comment:
Do we accept customer input `container_app_name` as the property name?

Reply:
It is optional, the customer can choose to specify it if they want. If they do, we make sure that and the arg name is consistent.